novawallet/Modules/Vote/Governance/TinderGov/Helpers/VoteAvailableReferendumSource.swift

```diff
@@ -0,0 +1,62 @@
+import SoraFoundation
+import Operation_iOS
+
+final class VoteAvailableReferendumSource {
+    var observers: [WeakWrapper] = []
+
+    private let referendumsSource: ReferendumObservableSourceProtocol
+    private let filter: ReferendumFilter.VoteAvailableChanges
+
+    private var changes: [ReferendumIdLocal: DataProviderChange<ReferendumLocal>] = [:]
+
+    private let mutex = NSLock()
+
+    init(
+        referendumsSource: ReferendumObservableSourceProtocol,
+        filter: ReferendumFilter.VoteAvailableChanges
+    ) {
+        self.referendumsSource = referendumsSource
+        self.filter = filter
+
+        referendumsSource.observe(self)
+    }
+}
+
+// MARK: ReferendumObservableSourceProtocol
+
+extension VoteAvailableReferendumSource: ReferendumObservableSourceProtocol, WeakWrappersClearing {
+    func observe(_ target: any ReferendumsSourceObserver) {
+        mutex.lock()
+        defer { mutex.unlock() }
+
+        clearEmptyWrappers()
+
+        let weakWrapper = WeakWrapper(target: target)
+        observers.append(weakWrapper)
+
+        guard !changes.isEmpty else { return }
+
+        target.didReceive(Array(changes.values))
```

Comment:
So, here might be updates and we consider only inserts in the TinderGovPresenter